### PR TITLE
Add querystring showing UI to allow forcing modes

### PR DESCRIPTION
**Pull request by kevinrobinson**
This adds a query string param that will show a UI allowing the user to set the strategy for picking audio or text responses.  For now it's either randomly mixed or all of a particular response type.

I also spent some time trying to get full renders working during test.  This involved:
- making a new `TestAuthContainer` to inject authentication context for test
- adding a call to `react-tap-event-plugin` in the Mocha `setup.js`
- running into a Material UI component relying on `element.scrollHeight`, which isn't supported in `jsdom`
- filing [this workaround upstream](https://github.com/callemall/material-ui/pull/5015) that I don't expect will be merged, and don't have any great ideas to workaround
- adding a minimal static HTML render test

### URL:
screen shot 2016-08-17 at 3 15 00 pm

### UI:
screen shot 2016-08-17 at 3 14 55 pm

### CODE REVIEW

File: ui/src/message_popup/experience_page.jsx [hunk 1]
```diff
@@ -129,16 +125,33 @@ export default React.createClass({
...
         />
-        {!hasStarted && this.renderInstructions()}
-        {hasStarted && questionsAnswered >= sessionLength && this.renderDone()}
-        {hasStarted && questionsAnswered < sessionLength && (_.has(this.props.query, 'mobilePrototype') ? this.renderMobilePrototype() : this.renderPopupQuestion())}
+        {this.renderMainScreen()}
       </div>
     );
   },
 
+  renderMainScreen() {
```
Comment (kevinrobinson) [inline ui/src/message_popup/experience_page.jsx]: Unrelated cleanup

File: ui/src/message_popup/experience_page.jsx [hunk 2]
```diff
@@ -159,16 +172,19 @@ export default React.createClass({
 
   renderInstructions() {
     const {scaffolding} = this.state;
+    const {query} = this.props;
+
     return (
       <VelocityTransitionGroup enter={{animation: "callout.pulse", duration: 500}} leave={{animation: "slideUp"}} runOnMount={true}>
-        <div>
+        <div className="instructions">
```
Comment (kevinrobinson) [inline ui/src/message_popup/experience_page.jsx]: @david-rosales This isn't ideal but since I couldn't get the `mount` deep renderer working yet, this is how you can add in classNames to check that the output of render is what you'd expect.

File: ui/src/message_popup/experience_page_test.jsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    </MuiThemeProvider>
+  );
+}
+
+// TODO(kr) it doesn't yet work to do a full render of ExperiencePage,
+// because the TextField Material UI component relies on element.scrollHeight,
+// which isn't present in jsdom.
+// Not sure of a good workaround, but submitted
+// https://github.com/callemall/material-ui/pull/5015 as a workaround and
+// to get input.
```
Comment (kevinrobinson) [inline ui/src/message_popup/experience_page_test.jsx]: 😞

**Comment by kevinrobinson** [comment]
@david-rosales Can you review por favor?

File: ui/src/message_popup/experience_page_test.jsx (hunk 3 above)
Comment (kevinrobinson) [inline ui/src/message_popup/experience_page_test.jsx]: yay for oviertassinari 😄

File: ui/src/message_popup/scaffolding_card.jsx [hunk 4]
```diff
@@ -99,7 +119,7 @@ export default React.createClass({
   },
   
   render(){
-    const {itemsToShow} = this.props;
+    const {itemsToShow, allowChoosingResponseMode} = this.props;
```
Comment (david-rosales) [inline ui/src/message_popup/scaffolding_card.jsx]: I wonder if we should make it so that `allowChoosingResponseMode` is enabled when the query string 'all' is present as well.

File: ui/src/message_popup/experience_page.jsx [hunk 5]
```diff
@@ -159,16 +172,19 @@ export default React.createClass({
...
           <InstructionsCard 
-           itemsToShow={this.props.query}
+           itemsToShow={query}
            />
           <ScaffoldingCard
             initialEmail={this.context.auth.userProfile.email}
             scaffolding={scaffolding}
-            itemsToShow={this.props.query}
+            itemsToShow={query}
+            allowChoosingResponseMode={_.has(query, 'modes')}
```
Comment (david-rosales) [inline ui/src/message_popup/experience_page.jsx]: Could we handle this inside of `ScaffoldingCard` because `itemsToShow` already contains `query`?

**Comment by david-rosales** [comment]
@kevinrobinson Looks good, approved!

File: ui/src/message_popup/scaffolding_card.jsx (hunk 4 above)
Comment (kevinrobinson) [inline ui/src/message_popup/scaffolding_card.jsx]: Cool, that makes sense with moving the check into `ScaffoldingCard`, updated!

**Comment by kevinrobinson** [comment]
@david-rosales Thanks for the great feedback and review! 👍
